### DISCUSSION TITLE
Can I ask why this deprecation notice is here? Is the plan to stop supporting replaceWordWith?

### DISCUSSION THREAD

**Discussion by dallanmc** [verronpro/office-stamper#86] [General]
https://github.com/verronpro/docx-stamper/blob/6e0a57e99773927d343c2e3608cf14c1f092b505/src/main/java/org/wickedsource/docxstamper/processor/replaceExpression/ReplaceWithProcessor.java#L36

**Comment by caring-coder** [comment]
Yea, it's a warning added by @ynaciri. 

The question about deprecation was brought up when he added the acess to config to all comment processors.
The main argument was about why should we support this particular processor, when the main placeholder actually have the same effect.

So the warning is there mostly because we might not do all the update to that specific processor in the future, and/or to start a conversation about its usage (thank you for starting it !)

**Reply by ynaciri** [reply]
Thanks for the reply @josephverron !

As he stated earlier, the replaceWordWith comment processor seems to do the same thing as having placeholders directly in the text.

It looks like a duplicate feature, except if there is a use case that usual placeholders won't answer.

**Comment by dallanmc** [comment]
I guess my use case is that my users do **everything** using comments only. The document itself contains absolutely no logic whatsoever, so the user has a good idea what the final doc will look like (without seeing placeholders).

It means I just have to educate my users on how to create comments instead of messing with the main body of the document.

**Reply by StefanLobbenmeierObjego** [reply]
yeah that is actually a really nice use case, because you can take an existing document and without changing any of the text in it turn it into a document template. This is also especially useful when the variable names are longer than their content, it is really hard to fit those long variable names in narrow tables, so replaceWith comes in very useful there to allow templating without messing up the whole table in the template

**Reply by caring-coder** [reply]
I agree with a target of being able to only comment an existing document is a nice use case. I'll try to make it clear that it is one of the goal of the project then.

**Comment by StefanLobbenmeierObjego** [comment]
Hey - I saw that replaceWordWith was now removed in release 3.5 via commit 0757e0f4

If someone wants to restore so you do not have to change your templates, here is what I did:

```kotlin
import io.github.oshai.kotlinlogging.KotlinLogging
import pro.verron.officestamper.api.CommentProcessor
import pro.verron.officestamper.api.ProcessorContext
import pro.verron.officestamper.preset.processors.replacewith.ReplaceWithProcessor

class ReplaceWordWithProcessor(val processorContext: ProcessorContext) :
    CommentProcessor(processorContext) {

    private var replaceWithProcessor: ReplaceWithProcessor = ReplaceWithProcessor(processorContext)

    @Suppress("unused")
    fun replaceWordWith(expression: String) {
        logger.warn { "replaceWordWith is deprecated, use replaceWith instead." }

        replaceWithProcessor.replaceWith(expression)
    }

    companion object {
        private val logger = KotlinLogging.logger {}
    }
}
```


And then you can just add it to the configuration like this:

```kotlin
        val configuration = OfficeStamperConfigurations.standard()
        configuration.commentProcessors[ReplaceWordWithProcessor::class.java] =
            CommentProcessorFactory(::ReplaceWordWithProcessor)
```
